packages/next/src/server/require-hook.ts

```diff
@@ -23,8 +23,6 @@ addHookAliases([
   ['styled-jsx', require.resolve('styled-jsx')],
   ['styled-jsx/style', require.resolve('styled-jsx/style')],
   ['styled-jsx/style', require.resolve('styled-jsx/style')],
-  ['server-only', require.resolve('next/dist/compiled/server-only')],
-  ['client-only', require.resolve('next/dist/compiled/client-only')],
```

Comment:
This is flawed because the require hook is executed in a normal Node.js process, not during build so there's no server/client layer involved. So that `require.resolve` will always give the path of the default export.

Reply:
Thanks for fixing this